src/JSDomEnvironment.js

```diff
@@ -104,8 +107,17 @@ JSDomEnvironment.prototype.dispose = function() {
   //this.global.close();
 };
 
+/**
+ * Evaluates the given source text as if it were in a file with the given name
+ * and returns the result.
+ */
 JSDomEnvironment.prototype.runSourceText = function(sourceText, fileName) {
-  return this.global.run(sourceText, fileName);
+  // TODO: Stop using the private API and instead configure jsdom with a
+  // resource loader and insert <script src="${filename}"> in the document. The
+  // reason we use vm for now is because the script element technique is slow.
+  return vm.runInContext(sourceText, this.document._ownerDocument._global, {
```

Comment:
Yeah we're discussing this in the rest of the PR. This particular technique is very very bad and will break easily in patch jsdom versions. Never access underscore-prefixed properties :(.